jep/0000/README.adoc

```diff
@@ -0,0 +1,187 @@
+
+==== Make default configuration secure
+
+Provide secure configuration as the default option.
+
+==== Make backup of Jenkins jobs history
+
+Backup Jenkins jobs history using tools provided by Kubernetes itself and optionally a cloud provider.
+
+==== Make errors more visible for end users
+
+Validate and visualise problems in configuration provided by the user.
+
+==== Reduce Jenkins master unavailability
+
+== Motivation
+
+Jenkins Operator contains complex knowledge about operating Jenkins, in particular running it on Kubernetes.
+Given that it shouldn't be a part Jenkins implementation.
+
+== Reasoning
+
+This document does not propose a specific design or implementation, as other JEPs might. This is intentional.
+This document specifies the concept of "Jenkins Operator" and the mission and priorities of the project.
+Later related JEPs will address design and implementation. Limiting the scope of this JEP to high-level goals will let
+us build a solid consensus for that direction separate from the design of individual features.
+
+== Backwards Compatibility
+
+Jenkins Operator is intended to run as an separate program and not require dedicated extension integrated in
+jenkins-core nor specific API implemented by plugins.
+
+=== CRD schema changes
+
+The specific version of Jenkins Operator supports a specific version of the Jenkins configuration saved in the CRD.
+Jenkins operator will inform the user of any inconsistencies contained in the CR for the version of the CRD he supports.
+Changing the Jenkins CRD configuration version requires changing the Jenkins Operator version.
+The migration process from one version to another will require from user to manually rewrite the Jenkins configuration
+to the new version and changes to the Jenkins Operator version.
+
+== Security
+
+Every sensitive information will be stored as link:https://kubernetes.io/docs/concepts/configuration/secret/[Kubernetes secrets].
+Only Jenkins Operator and Jenkins master will have access to those secrets. Access to secrets will be secured by
+link:https://kubernetes.io/docs/reference/access-authn-authz/rbac/[Kubernetes RBAC policies] and
+link:https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/[Kubernetes Service Accounts].
+Communication between Jenkins Operator and Jenkins master will be secured using TLS.
+
+== Infrastructure Requirements
+
+The infrastructure dependencies for the `virtuslab/jenkins-operator` image are mostly on
+services external to the Jenkins project such as link:https://hub.docker.com[Docker Hub].
+
+== Testing
+
+Testing is performed within the context of the existing automated testing in the
+link:https://github.com/VirtusLab/jenkins-operator[VirtusLab/jenkins-operator]
+repository. This includes the expected amount of unit tests, and end-to-end tests which run Jenkins Operator
+and Jenkins on a link:https://github.com/kubernetes/minikube[Minikube] Kubernetes cluster.
```

Comment:
BTW I have found Microk8s quite nice, in case your build host is a Linux VM.

Reply:
@jglick From my experience Minikube is more mature project and you can run it directly on host with `--vm-driver=none` parameter.